plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -2846,11 +3175,17 @@ public RepositoryDirectoryInterface findDirectory( ObjectId directory ) throws K
   public JobMeta loadJob( ObjectId idJob, String versionLabel ) throws KettleException {
     try {
       RepositoryFile file = null;
-      if ( versionLabel != null ) {
-        file = pur.getFileAtVersion( idJob.getId(), versionLabel );
-      } else {
-        file = pur.getFileById( idJob.getId() );
+      readWriteLock.readLock().lock();
+      try {
+        if ( versionLabel != null ) {
+          file = pur.getFileAtVersion( idJob.getId(), versionLabel );
+        } else {
+          file = pur.getFileById( idJob.getId() );
+        }
+      } finally {
+        readWriteLock.readLock().unlock();
```

Comment:
The read lock should not be release between this and lower in the same method. The lock should be kept.